src/main/kotlin/no/nav/klage/oppgave/api/OppgaveFacade.kt

```diff
@@ -27,45 +21,22 @@ class OppgaveFacade(
         private val securelogger = getSecureLogger()
     }
 
-    fun searchOppgaver(oppgaverSearchCriteria: OppgaverSearchCriteria): OppgaverRespons {
-        if (unleash.isEnabled("klage.sokMedES")) {
-            val esResponse = elasticsearchRepository.findByCriteria(oppgaverSearchCriteria)
-            return OppgaverRespons(
-                antallTreffTotalt = esResponse.totalHits.toInt(),
-                oppgaver = oppgaveMapper.mapEsOppgaverToView(
-                    esResponse.searchHits.map { it.content },
-                    oppgaverSearchCriteria.isProjectionUtvidet()
-                )
-            )
-        } else {
-            val oppgaveResponse = oppgaveService.searchOppgaver(oppgaverSearchCriteria)
-            return OppgaverRespons(
-                antallTreffTotalt = oppgaveResponse.antallTreffTotalt,
-                oppgaver = oppgaveMapper.mapOppgaverToView(
-                    oppgaveResponse.oppgaver,
-                    oppgaverSearchCriteria.isProjectionUtvidet()
-                )
-            )
-        }
-    }
-
-    fun assignOppgave(oppgaveId: Long, saksbehandlerIdent: String?, oppgaveVersjon: Int?) {
-        oppgaveService.assignOppgave(oppgaveId, saksbehandlerIdent, oppgaveVersjon)
-    }
-
     fun saveOppgaveKopi(oppgave: OppgaveKopiAPIModel) {
-        oppgaveKopiService.saveOppgaveKopi(oppgaveMapper.mapOppgaveKopiAPIModelToOppgaveKopi(oppgave))
-        if (unleash.isEnabled("klage.indekserMedES")) {
-            try {
-                elasticsearchRepository.save(oppgaveMapper.mapOppgaveKopiAPIModelToEsOppgave(oppgave))
-            } catch (e: Exception) {
-                if (e.message?.contains("version_conflict_engine_exception") == true) {
-                    logger.info("Later version already indexed, ignoring this..")
-                } else {
-                    logger.error("Unable to index OppgaveKopi, see securelogs for details")
-                    securelogger.error("Unable to index OppgaveKopi", e)
-                }
+        val klagebehandlingerOgMottak =
+            oppgaveKopiService.saveOppgaveKopi(oppgaveMapper.mapOppgaveKopiAPIModelToOppgaveKopi(oppgave))
+
+        try {
+            klagebehandlingerOgMottak.forEach { elasticsearchRepository.save(oppgaveMapper.mapToEs(it)) }
```

Comment:
Skjønner jeg rett nå? Når vi mottar en oppgave så henter vi ut alle versjoner vi har for oppgaven og indekserer de på nytt i ES?

Reply:
Vi henter ut alle versjonene, fordi vi trenger historikken for å finne alle dataene vi trenger for å oppdatere Mottak. Men vi kan jo ha flere mottak koblet til en oppgave, så det er derfor det er "klagebehandlingerOgMottak.forEach" her. Det skal vel ikke skje, men teoretisk kan vi ha flere åpne mottak. Så jeg forholdt meg til teorien her.. :)